internal/locate/region_request.go

```diff
@@ -397,12 +398,25 @@ func (state *tryFollower) next(bo *retry.Backoffer, selector *replicaSelector) (
 		selector.invalidateRegion()
 		return nil, nil
 	}
-	return selector.buildRPCContext(bo)
+	rpcCtx, err := selector.buildRPCContext(bo)
+	if err != nil || rpcCtx == nil {
+		return rpcCtx, err
+	}
+	if state.fallbackFromLeader {
+		replicaRead := true
+		rpcCtx.contextPatcher.replicaRead = &replicaRead
+	}
+	return rpcCtx, err
 }
 
 func (state *tryFollower) onSendSuccess(selector *replicaSelector) {
-	if !selector.region.switchWorkLeaderToPeer(selector.targetReplica().peer) {
```

Comment:
The former naming and meaning of the `switchWorkLeaderToPeer` function is quite confusing, I don't understand what's the purpose of it..

Reply:
The former usage of `tryFollower` is after the failure of `accessKnownLeader`, in this case, if one of the follower can serve the leader-read request, it's the new leader, so switch the leader to this peer.